src/libs/elektra/keyset.c

```diff
@@ -380,7 +384,15 @@ int ksDel (KeySet * ks)
 int ksClear (KeySet * ks)
 {
 	ksClose (ks);
-	// ks->array empty now
+// ks->array empty now
+
+#ifdef ELEKTRA_ENABLE_OPTIMIZATIONS
+	ks->opmphm = opmphmNew ();
+	if (!ks->opmphm)
+	{
+		return -1;
```

Comment:
Is it sensible to continue here? (If there was enough memory for array, but not for opmphm? Array should be allocated first, though)

Btw. the current implementation of opmphmNew uses Malloc although Calloc would be better suitable?